src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -431,7 +431,10 @@ private DeclarationModifiers MakeModifiers(TypeKind typeKind, BindingDiagnosticB
                         case SymbolKind.Namespace:
                             for (var i = 1; i < partCount; i++)
                             {
-                                diagnostics.Add(ErrorCode.ERR_DuplicateNameInNS, declaration.Declarations[i].NameLocation, this.Name, this.ContainingSymbol);
+                                // note: a declaration with the 'file' modifier will only be grouped with declarations in the same file.
+                                diagnostics.Add((result & DeclarationModifiers.File) != 0
+                                    ? ErrorCode.ERR_FileLocalDuplicateNameInNS
```

Comment:
Are we testing this case? (It looks like the cases where we test the new error do not involve `partial` types.)

Reply:
I think 'Duplication_06' tests this case.

Reply:
Mostly there are existing tests for `partial` in the same file and different files. `Duplication_06` changed, but mostly I think the baselines didn't change because the missing `partial` error takes priority over other duplication errors.

Reply:
Thanks. I asked because I was looking at commit 1 where `Duplication_06` was still referencing the original `ErrorCode`.